UnoCheck/Checkups/PSExecutionPolicyCheckup.cs

```diff
@@ -0,0 +1,42 @@
+﻿using System.Threading.Tasks;
+
+using DotNetCheck.Models;
+using DotNetCheck.Solutions;
+
+namespace DotNetCheck.Checkups
+{
+	internal class PSExecutionPolicyCheckup : Checkup
+	{
+		public override string Id => "psexecpolicy";
+
+		public override string Title => $"PowerShell Execution Policy";
+
+		public override bool IsPlatformSupported(Platform platform) => platform == Platform.Windows;
+
+		public override Task<DiagnosticResult> Examine(SharedState state)
+		{
+			var result = ShellProcessRunner.Run("powershell", "-Command Get-ExecutionPolicy -Scope CurrentUser");
+
+			if (result.Success)
+			{
+				if (result.StandardOutput.Count != 0 && result.StandardOutput[0] == "RemoteSigned")
+				{
+					return Task.FromResult(DiagnosticResult.Ok(this));
+				}
+				else
+				{
+					return Task.FromResult(
+						new DiagnosticResult(
+							Status.Error,
+							this,
+							"PowerShell execution policy for current user is not RemoteSigned.",
+							new("Change PowerShell execution policy to RemoteSigned", new PSExecutionPolicySolution())));
+				}
+			}
+			else
+			{
+				return Task.FromResult(new DiagnosticResult(Status.Error, this, "Failed to get PowerShell execution policy."));
+			}
```

Comment:
I'm wondering why you're not using the dotnet powershell API instead of calling the process like that?

https://learn.microsoft.com/en-us/dotnet/api/system.management.automation.powershell.invokeasync?view=powershellsdk-7.4.0#system-management-automation-powershell-invokeasync

```csharp
using (PowerShell ps = PowerShell.Create())
{
    ps.AddCommand("Get-ExecutionPolicy").AddParameter("Scope", "CurrentUser");

    var result = await ps.InvokeAsync();
    
    [...]

}
```

Reply:
Three reasons:
1 - ShellProcessRunner is used throughout Uno.Check
2 - Hosting Pwsh makes no sense here. See: https://devblogs.microsoft.com/powershell/depending-on-the-right-powershell-nuget-package-in-your-net-project
3 - I don't want to take a dependency on it